lib/src/main/java/ca/jbrains/pos/PointOfSale.java

```diff
@@ -7,20 +7,20 @@
 
 public class PointOfSale {
     public static void main(String[] args) {
-        parseInput(new InputStreamReader(System.in)).forEachOrdered(
-                line -> displayToConsole(displaySellOneItem(line, (ignored) -> "::a hardcoded response for every barcode::"))
+        readBarcodeFrom(new InputStreamReader(System.in)).forEachOrdered(
+                barcodeInput -> displayToConsole(findProductPrice(barcodeInput, (ignored) -> "::a hardcoded response for every barcode::"))
         );
     }
 
     private static void displayToConsole(String message) {
         System.out.println(message);
     }
 
-    public static Stream<String> parseInput(Reader simulateInputFromStdin) {
-        return new BufferedReader(simulateInputFromStdin).lines();
+    public static Stream<String> readBarcodeFrom(Reader inputReader) {
+        return new BufferedReader(inputReader).lines();
     }
 
-    public static String displaySellOneItem(String barcode, SaleController saleController) {
+    public static String findProductPrice(String barcode, SaleController saleController) {
```

Comment:
- I think the intention here is givenBarCode (unparsed, so it could be valid or invalid) and then we find the price for the given barCode, hence, find the productPrice is a better name compare to displaySellOneItem
- I feel uncomfortable that findProductPrice doesn't actually express the intent that we want to sellOneItem, and I'm not sure what's the better way to express the intent that we want to sellOneItem  (instead of findProductPrice).
- Howver, I feel like findProductPrice is a better name compare to displaySellOneItem, but it doesn't express the intent from the system that we want to sell1item.

Reply:
Indeed, "find product price" describes _one part_ of what this function does. It is _accurate but vague_. Think about a more-precise name.

Reply:
"input reader" is _accurate but vague_. Since arguments are always (at least) input, I would remove the duplication and rename to just `reader`.

But then... reader of what? For now, only barcodes, so `barcodesReader` might be fine.

Now imagine 3 features into the future, when we still read from that reader, but not only barcodes. What might you call it then?